File: day2/day_2_part_1.py
with open("input.txt", "r") as f:
    data = f.readlines()

max_colors ={
    'red': 12,
    'green': 13,
    'blue': 14
}

total = 0

for line in data:
    possible = True
    for color in max_colors:
        color_indexes = [i for i in range(len(line)) if line.startswith(color, i)]
        for i in color_indexes:
            number_of_color = line[i-3] + line[i-2]
            if int(number_of_color) > max_colors[color]:
                possible = False
                break
    if possible:
        first_index = line.find(' ')+1
        last_index = line.find(':')

        total += int(line[first_index:last_index])
    else:
        possible = True


print(total)

File: day2/day2Part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	file, err := os.ReadFile("./test.txt")
	if err != nil {
		fmt.Println("Fuck: ", err)
		return
	}
	defer file.Close()
	maxColors := map[string]int{
		"red":   12,
		"green": 13,
		"blue":  14,
	}
	scanner := bufio.NewScanner(file)
	total := 0
	for scanner.Scan() {
		line := scanner.Text()
	}

	fmt.Println("Hello World.")
}
